python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -434,26 +436,47 @@ def passes(
         context: AssetConditionEvaluationContext,
         asset_partitions: Iterable[AssetKeyPartitionKey],
     ) -> Iterable[AssetKeyPartitionKey]:
+        from dagster._core.storage.event_log.sql_event_log import get_max_event_records_limit
+
         if self.latest_run_required_tags is None:
             return asset_partitions
 
         will_update_asset_partitions: Set[AssetKeyPartitionKey] = set()
+        storage_ids_to_fetch_by_key: Dict[AssetKey, List[int]] = defaultdict(list)
 
-        asset_partitions_by_latest_run_id: Dict[str, Set[AssetKeyPartitionKey]] = defaultdict(set)
         for asset_partition in asset_partitions:
             if context.will_update_asset_partition(asset_partition):
                 will_update_asset_partitions.add(asset_partition)
             else:
-                record = context.instance_queryer.get_latest_materialization_or_observation_record(
-                    asset_partition
+                latest_storage_id = (
+                    context.instance_queryer.get_latest_materialization_or_observation_storage_id(
+                        asset_partition=asset_partition
+                    )
                 )
+                if latest_storage_id is not None:
+                    storage_ids_to_fetch_by_key[asset_partition.asset_key].append(latest_storage_id)
 
-                if record is None:
-                    raise RuntimeError(
-                        f"No materialization record found for asset partition {asset_partition}"
-                    )
+        asset_partitions_by_latest_run_id: Dict[str, Set[AssetKeyPartitionKey]] = defaultdict(set)
 
-                asset_partitions_by_latest_run_id[record.run_id].add(asset_partition)
+        step = get_max_event_records_limit()
```

Comment:
i'm seeing some indication in my testing that this step limit may be a little high (hitting some 15 second timeouts when there are actually >10k events to fetch, e.g. in a large empty-cursor run) - in real life the timeout is higher than 15 seconds, but we may still want more headroom here